### ISSUE TITLE
Traffic monitoring using YOLO11

### ISSUE THREAD

**Issue by caisamuels** [ultralytics/ultralytics#18593]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have been tasked with creating a traffic monitoring system using computer vision which classifies vehicles and estimates speed. This data will then be fed into a web dashboard displaying live visualisations. I was originally going to run YOLO11 on a Raspberry Pi 3B, however, it became clear that this would not work due to hardware limitations. I now plan on streaming the camera feed from the Raspberry Pi to a machine with a high-spec GPU. What would be the best way to go about this project?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @caisamuels, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for detailed guidance, where you can also explore [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions are addressed there and may assist in your traffic monitoring system project.

If this is a custom training ❓ Question, please provide additional context, such as your dataset description, vehicle classification labels, speed estimation methodology, and any potential training logs. Additionally, ensure you follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize performance.

For building scalable pipelines, you may find our in-depth discussions on [Discourse](https://community.ultralytics.com) or real-time chats in [Discord](https://discord.com/invite/ultralytics) helpful 🎧. You can also exchange ideas with the wider Ultralytics community on our [Subreddit](https://reddit.com/r/Ultralytics).

## Suggested Steps

1. If hardware limitations affect your implementation, assess the possibility of streaming video via protocols like RTSP from the Raspberry Pi to your GPU-equipped machine.
2. Use the [Ultralytics YOLO Python API](https://docs.ultralytics.com/usage/python/) to ingest live streams, process them for object classification, and extract relevant metrics. 
3. Explore setting up a lightweight dashboard via frameworks like FastAPI or Flask to serve visualizations.
4. Train models specifically tuned for vehicle types and speed estimation. Utilize transfer learning, starting with a pretrained YOLO11 model, and fine-tune using a dataset relevant to your traffic scenes.

## Upgrade

Ensure you're using the latest `ultralytics` package in an environment meeting all dependencies, including [Python>=3.8](https://www.python.org/) and [PyTorch>=1.8](https://pytorch.org/get-started/locally/). Update your software by running the following:

```bash
pip install -U ultralytics
```

Check the installation instructions: [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml).

## Environments

For faster prototyping or model testing, YOLO11 can run in verified environments with preinstalled dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** and **AWS** GPU instances: Follow our [GCP Quickstart](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) or [AWS Quickstart](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the above badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, verifying stable operation across macOS, Windows, and Ubuntu platforms.

This is an automated response 🤖. An Ultralytics engineer will review your query and get back to you soon. 🚦

**Comment by glenn-jocher** [comment]
@caisamuels thank you for your question. To create your traffic monitoring system, streaming the camera feed from the Raspberry Pi to a high-spec GPU machine is a viable approach given the Raspberry Pi 3B's hardware constraints. Use a protocol like RTSP or gRPC for efficient streaming. Run YOLO11 on the GPU machine for vehicle classification and speed estimation, leveraging the `SpeedEstimator` functionality. For exporting data to your web dashboard, consider integrating with a message broker like MQTT or a database to store and visualize results. Refer to the [Speed Estimation Guide](https://docs.ultralytics.com/guides/speed-estimation/) for implementation details and best practices. Let us know if you need further assistance!
